### ISSUE TITLE
Undefined offset: 0 ErrorException

### ISSUE THREAD

**Issue by userbig** [eveseat/eseye#44]
Hi.

When I'm trying to get corporation wallet data I receive an exception
```

 ErrorException  : Undefined offset: 0

  at D:\php\eve\vendor\symfony\var-dumper\Cloner\VarCloner.php:83
    79|             }
    80|             foreach ($vals as $k => $v) {
    81|                 // $v is the original value or a stub object in case of hard references
    82|                 $refs[$k] = $cookie;
  > 83|                 if ($zvalIsRef = $vals[$k] === $cookie) {
    84|                     $vals[$k] = &$stub;         // Break hard references to make $queue completely
    85|                     unset($stub);               // independent from the original structure
    86|                     if ($v instanceof Stub && isset($hardRefs[\spl_object_id($v)])) {
    87|                         $vals[$k] = $refs[$k] = $v;

  Exception trace:

  1   Symfony\Component\VarDumper\Cloner\AbstractCloner::Symfony\Component\VarDumper\Cloner\{closure}("Undefined offset: 0", "D:\php\eve\vendor\symfony\var-dumper\Cloner\VarCloner.php")
      D:\php\eve\vendor\symfony\var-dumper\Cloner\VarCloner.php:83

  2   Symfony\Component\VarDumper\Cloner\VarCloner::doClone(Object(Seat\Eseye\Containers\EsiResponse))
      D:\php\eve\vendor\symfony\var-dumper\Cloner\AbstractCloner.php:232

  Please use the argument -v to see more details.

```
Code: 
```
$data =  $this->esi->invoke('get', '/corporations/{corporation_id}/wallets', [
            'corporation_id' => $corporation_id,
        ]);
 dd($data);
```
All character invoke calls works fine. And ofc i can get this data by parsing raw json or doing collect($data)

**Comment by warlof** [comment]
Hi @userbig,

can you provide the complete stack related to this error ?

Symfony isn't part of our packages, so either there is an issue with it and we can't do anything unfortunately.
Or the issue is upper in the stack extract you're providing :p

Did you tried with the missing trailing slash in your endpoint ? `/corporations/{corporation_id}/wallets/`
Also, are you sure you don't have any global version which is altered prior this call and that could throw an exception ? (this endpoint is only available on `v1`)

**Comment by userbig** [comment]
```

   ErrorException  : Undefined offset: 0

  at D:\php\eve\vendor\symfony\var-dumper\Cloner\VarCloner.php:83
    79|             }
    80|             foreach ($vals as $k => $v) {
    81|                 // $v is the original value or a stub object in case of hard references
    82|                 $refs[$k] = $cookie;
  > 83|                 if ($zvalIsRef = $vals[$k] === $cookie) {
    84|                     $vals[$k] = &$stub;         // Break hard references to make $queue completely
    85|                     unset($stub);               // independent from the original structure
    86|                     if ($v instanceof Stub && isset($hardRefs[\spl_object_id($v)])) {
    87|                         $vals[$k] = $refs[$k] = $v;

  Exception trace:

  1   Symfony\Component\VarDumper\Cloner\AbstractCloner::Symfony\Component\VarDumper\Cloner\{closure}("Undefined offset: 0", "D:\php\eve\vendor\symfony\var-dumper\Cloner\VarCloner.php")
      D:\php\eve\vendor\symfony\var-dumper\Cloner\VarCloner.php:83

  2   Symfony\Component\VarDumper\Cloner\VarCloner::doClone(Object(Seat\Eseye\Containers\EsiResponse))
      D:\php\eve\vendor\symfony\var-dumper\Cloner\AbstractCloner.php:232

  3   Symfony\Component\VarDumper\Cloner\AbstractCloner::cloneVar(Object(Seat\Eseye\Containers\EsiResponse))
      D:\php\eve\vendor\symfony\var-dumper\VarDumper.php:34

  4   Symfony\Component\VarDumper\VarDumper::Symfony\Component\VarDumper\{closure}(Object(Seat\Eseye\Containers\EsiResponse))
      D:\php\eve\vendor\symfony\var-dumper\VarDumper.php:38

  5   Symfony\Component\VarDumper\VarDumper::dump(Object(Seat\Eseye\Containers\EsiResponse))
      D:\php\eve\vendor\symfony\var-dumper\Resources\functions\dump.php:37

  6   dd(Object(Seat\Eseye\Containers\EsiResponse))
      D:\php\eve\app\UserDie\ESI\ESI.php:55

  7   UserDie\ESI\ESI::getCorporationJournalByDivision()
      D:\php\eve\app\UserDie\ESI\ESI.php:67

  8   UserDie\ESI\ESI::getReceipt()
      D:\php\eve\app\UserDie\ESI\Controllers\Journal.php:30

  9   UserDie\ESI\Controllers\Journal::mapReceipts()
      D:\php\eve\app\Console\Commands\getReceipts.php:44

  10  App\Console\Commands\getReceipts::handle()
      D:\php\eve\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:29

  11  call_user_func_array([])
      D:\php\eve\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:29

  12  Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
      D:\php\eve\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:87

  13  Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Object(Closure))
      D:\php\eve\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:31

  14  Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), [])
      D:\php\eve\vendor\laravel\framework\src\Illuminate\Container\Container.php:564

  15  Illuminate\Container\Container::call()
      D:\php\eve\vendor\laravel\framework\src\Illuminate\Console\Command.php:179

  16  Illuminate\Console\Command::execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
      D:\php\eve\vendor\symfony\console\Command\Command.php:255

  17  Symfony\Component\Console\Command\Command::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
      D:\php\eve\vendor\laravel\framework\src\Illuminate\Console\Command.php:166

  18  Illuminate\Console\Command::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\php\eve\vendor\symfony\console\Application.php:888

  19  Symfony\Component\Console\Application::doRunCommand(Object(App\Console\Commands\getReceipts), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console
\Output\ConsoleOutput))
      D:\php\eve\vendor\symfony\console\Application.php:264

  20  Symfony\Component\Console\Application::doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\php\eve\vendor\symfony\console\Application.php:145

  21  Symfony\Component\Console\Application::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\php\eve\vendor\laravel\framework\src\Illuminate\Console\Application.php:89

  22  Illuminate\Console\Application::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\php\eve\vendor\laravel\framework\src\Illuminate\Foundation\Console\Kernel.php:122

  23  Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\php\eve\artisan:37

```

> Did you tried with the missing trailing slash in your endpoint ? `/corporations/{corporation_id}/wallets/`
Also, are you sure you don't have any global version which is altered prior this call and that could throw an exception ? (this endpoint is only available on `v1`)

There no problem with endpoint, version or url. There problem with dd(). I dont know why, but this helper cant iterate eseye corp wallet return instance.

**Comment by leonjza** [comment]
So weird, does a simple `var_dump()` work?

**Comment by userbig** [comment]
Yup. Works.

**Comment by drkthunder02** [comment]
So for the following 2 lines you need to make sure the memory of your arrays contains the valid values you think it should have, and these two lines are:
```php
 82|                 $refs[$k] = $cookie;
 83|                 if ($zvalIsRef = $vals[$k] === $cookie) {
```

Instead of $vals[$k] try to just reference $v.  Since you are in a foreach loop $vals[$k] = $v as defined by the start of the foreach($vals as $k => $v).  It seems like the compiler is throwing a fit trying to reference the $vals parameter within the foreach loop but it doesn't exist in the foreach loop due to how it's declared.  It's a bit hard to explain over text, and is more easily done on a whiteboard.

TL:DR:  Change line 83 to read if($zvallsRef = $v === $cookie) {

**Comment by leonjza** [comment]
`var-dumper` is a dependency that Eseye does not include by itself. I suspect you may have this as part of the larger project you are using Eseye in, and it may mean you need to update the dependency.

That being said, could you try and give use the version you are using (and possibly try and update the dependency? `composer show -i | grep -i dump` may help you determine the installed version.

**Comment by userbig** [comment]
```
dnoegel/php-xdg-base-dir              0.1      implementation of xdg base directory specification for php
doctrine/inflector                    v1.3.0   Common String Manipulations with regard to casing and singular/plural rules.
doctrine/instantiator                 1.1.0    A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                        v1.0.1   Base library for a lexer that can be used in Top-Down, Recursive Descent Parsers.
dragonmantank/cron-expression         v2.2.0   CRON for PHP: Calculate the next or previous run date and determine if a CRON expression is due
egulias/email-validator               2.1.6    A library for validating emails against several RFCs
erusev/parsedown                      1.7.1    Parser for Markdown.
eveseat/eseye                         1.1.6    A Standalone PHP ESI (EVE Swagger Interface) Client Library
fideloper/proxy                       4.0.0    Set trusted proxies for Laravel
filp/whoops                           2.2.1    php error handling for cool kids
fzaninotto/faker                      v1.8.0   Faker is a PHP library that generates fake data for you.
guzzlehttp/guzzle                     6.3.3    Guzzle is a PHP HTTP client library
guzzlehttp/promises                   v1.3.1   Guzzle promises library
guzzlehttp/psr7                       1.4.2    PSR-7 message implementation that also provides common utility methods
hamcrest/hamcrest-php                 v2.0.0   This is the PHP port of Hamcrest Matchers
irazasyed/telegram-bot-sdk            v2.2.0   The Unofficial Telegram Bot API PHP SDK
jakub-onderka/php-console-color       v0.2
jakub-onderka/php-console-highlighter v0.3.2
jaybizzle/crawler-detect              v1.2.67  CrawlerDetect is a PHP class for detecting bots/crawlers/spiders via the user agent
jaybizzle/laravel-crawler-detect      v1.2.0   A Laravel package to detect web crawlers via the user agent
jeremykenedy/laravel-logger           v2.4.1   Laravel Logger Package
laravel/framework                     v5.6.38  The Laravel Framework.
laravel/socialite                     v3.1.1   Laravel wrapper around OAuth 1 & OAuth 2 libraries.
laravel/tinker                        v1.0.7   Powerful REPL for the Laravel framework.
laravelcollective/html                v5.6.10  HTML and Form Builders for the Laravel Framework
league/flysystem                      1.0.47   Filesystem abstraction: Many filesystems, one API.
league/oauth1-client                  1.7.0    OAuth 1.0 Client Library
mockery/mockery                       1.1.0    Mockery is a simple yet flexible PHP mock object framework
monolog/monolog                       1.23.0   Sends your logs to files, sockets, inboxes, databases and various web services
myclabs/deep-copy                     1.8.1    Create deep copies (clones) of your objects
nesbot/carbon                         1.25.0   A simple API extension for DateTime.
nikic/php-parser                      v4.0.4   A PHP parser written in PHP
nullx27/eveonline-socialite           v0.5     EveOnline SSO Provider for Laravel Socialite
nunomaduro/collision                  v2.0.3   Cli error handling for console/command-line PHP applications.
paragonie/random_compat               v9.99.99 PHP 5.x polyfill for random_bytes() and random_int() from PHP 7
phar-io/manifest                      1.0.3    Component for reading phar.io manifest information from a PHP Archive (PHAR)
phar-io/version                       2.0.1    Library for handling version information and constraints
phpdocumentor/reflection-common       1.0.1    Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock     4.3.0    With this component, a library can provide support for annotations via DocBlocks or otherwise retrieve information that is embedded i...
phpdocumentor/type-resolver           0.4.0
phpspec/prophecy                      1.8.0    Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage             6.0.7    Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator             2.0.2    FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template             1.2.1    Simple template engine.
phpunit/php-timer                     2.0.0    Utility class for timing
phpunit/php-token-stream              3.0.0    Wrapper around PHP's tokenizer extension.
phpunit/phpunit                       7.3.5    The PHP Unit Testing framework.
predis/predis                         v1.1.1   Flexible and feature-complete Redis client for PHP and HHVM
psr/container                         1.0.0    Common Container Interface (PHP FIG PSR-11)
psr/http-message                      1.0.1    Common interface for HTTP messages
psr/log                               1.0.2    Common interface for logging libraries
psr/simple-cache                      1.0.1    Common interfaces for simple caching
psy/psysh                             v0.9.8   An interactive shell for modern PHP.
ramsey/uuid                           3.8.0    Formerly rhumsaa/uuid. A PHP 5.4+ library for generating RFC 4122 version 1, 3, 4, and 5 universally unique identifiers (UUID).
sebastian/code-unit-reverse-lookup    1.0.1    Looks up which function or method a line of code belongs to
sebastian/comparator                  3.0.2    Provides the functionality to compare PHP values for equality
sebastian/diff                        3.0.1    Diff implementation
sebastian/environment                 3.1.0    Provides functionality to handle HHVM/PHP environments
sebastian/exporter                    3.1.0    Provides the functionality to export PHP variables for visualization
sebastian/global-state                2.0.0    Snapshotting of global state
sebastian/object-enumerator           3.0.3    Traverses array structures and object graphs to enumerate all referenced objects
sebastian/object-reflector            1.1.1    Allows reflection of object attributes, including inherited and non-public ones
sebastian/recursion-context           3.0.0    Provides functionality to recursively process PHP variables
sebastian/resource-operations         1.0.0    Provides a list of PHP built-in functions that operate on resources
sebastian/version                     2.0.1    Library that helps with managing the version number of Git-hosted PHP projects
swiftmailer/swiftmailer               v6.1.3   Swiftmailer, free feature-rich PHP mailer
symfony/console                       v4.1.5   Symfony Console Component
symfony/css-selector                  v4.1.5   Symfony CssSelector Component
symfony/debug                         v4.1.5   Symfony Debug Component
symfony/event-dispatcher              v4.1.5   Symfony EventDispatcher Component
symfony/finder                        v4.1.5   Symfony Finder Component
symfony/http-foundation               v4.1.5   Symfony HttpFoundation Component
symfony/http-kernel                   v4.1.5   Symfony HttpKernel Component
symfony/polyfill-ctype                v1.9.0   Symfony polyfill for ctype functions
symfony/polyfill-mbstring             v1.9.0   Symfony polyfill for the Mbstring extension
symfony/polyfill-php72                v1.9.0   Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/process                       v4.1.5   Symfony Process Component
symfony/routing                       v4.1.5   Symfony Routing Component
symfony/translation                   v4.1.5   Symfony Translation Component
symfony/var-dumper                    v4.1.5   Symfony mechanism for exploring and dumping PHP variables
theseer/tokenizer                     1.1.0    A small library for converting tokenized PHP source code into XML and potentially other formats
tijsverkoyen/css-to-inline-styles     2.2.1    CssToInlineStyles is a class that enables you to convert HTML-pages/files into HTML-pages/files with inline styles. This is very usef...
vlucas/phpdotenv                      v2.5.1   Loads environment variables from `.env` to `getenv()`, `$_ENV` and `$_SERVER` automagically.
webmozart/assert                      1.3.0    Assertions to validate method input/output with nice error messages.

```

I`m using laravel 5.6 in my project.
